brig/cmd/brig/commands/build_list.go

```diff
@@ -32,48 +38,88 @@ var buildList = &cobra.Command{
 		if len(args) > 0 {
 			proj = args[0]
 		}
-		return listBuilds(cmd.OutOrStdout(), proj)
+
+		c, err := kubeClient()
+		if err != nil {
+			return err
+		}
+
+		bls, err := getBuilds(proj, c, buildListCount)
+		if err != nil {
+			return err
+		}
+
+		listBuilds(bls, cmd.OutOrStdout())
+		return nil
 	},
 }
 
-func listBuilds(out io.Writer, project string) error {
-	c, err := kubeClient()
-	if err != nil {
-		return err
+func listBuilds(bs []*buildForStdout, out io.Writer) {
+	table := uitable.New()
+	table.AddRow("ID", "TYPE", "PROVIDER", "PROJECT", "STATUS", "AGE")
+	for _, b := range bs {
+		table.AddRow(b.ID, b.Type, b.Provider, b.ProjectID, b.status, b.since)
 	}
+	fmt.Fprintln(out, table)
+}
 
-	store := kube.New(c, globalNamespace)
+func getBuilds(project string, client kubernetes.Interface, count int) ([]*buildForStdout, error) {
 
-	var bs []*brigade.Build
+	store := kube.New(client, globalNamespace)
+
+	var builds []*brigade.Build
+	var err error
 	if project == "" {
-		bs, err = store.GetBuilds()
+		builds, err = store.GetBuilds()
 		if err != nil {
-			return err
+			return nil, err
 		}
 	} else {
 		proj, err := store.GetProject(project)
 		if err != nil {
-			return err
+			return nil, err
 		}
 
-		bs, err = store.GetProjectBuilds(proj)
+		builds, err = store.GetProjectBuilds(proj)
 		if err != nil {
-			return err
+			return nil, err
 		}
 	}
 
-	table := uitable.New()
-	table.AddRow("ID", "TYPE", "PROVIDER", "PROJECT", "STATUS", "AGE")
+	// sorting here on StartTime because we do not want to rely on K8s sorting (which would be the order that Secrets/Pods were created)
+	// remember that most recent *started* builds must be at the top
+	sort.Slice(builds, func(i, j int) bool {
+		if builds[i].Worker == nil || builds[j] == nil {
```

Comment:
Do we need a nil check on `builds[j].Worker` as well?  When running a list w/ count on a particular project and I see the following:
```
 $ bd build list --count 10 <project>
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1ac5606]

goroutine 1 [running]:
github.com/Azure/brigade/brig/cmd/brig/commands.getBuilds.func1(0x43, 0x0, 0x0)
	/Users/vdice/go/src/github.com/Azure/brigade/brig/cmd/brig/commands/build_list.go:95 +0x76
sort.doPivot_func(0xc000a15c10, 0xc0020874e0, 0x0, 0x10c, 0x0, 0x8)
	/usr/local/Cellar/go/1.11/libexec/src/sort/zfuncversion.go:119 +0x246
sort.quickSort_func(0xc000a15c10, 0xc0020874e0, 0x0, 0x10c, 0x12)
	/usr/local/Cellar/go/1.11/libexec/src/sort/zfuncversion.go:143 +0x9a
sort.Slice(0x1b388a0, 0xc0020874c0, 0xc000a15c10)
	/usr/local/Cellar/go/1.11/libexec/src/sort/slice.go:21 +0x12f
github.com/Azure/brigade/brig/cmd/brig/commands.getBuilds(0x7ffeefbfe705, 0x10, 0x1e4de00, 0xc000274a50, 0xa, 0xc0000eca00, 0x1ce336f, 0x4, 0x1ce329b, 0x4)
	/Users/vdice/go/src/github.com/Azure/brigade/brig/cmd/brig/commands/build_list.go:91 +0x14d
github.com/Azure/brigade/brig/cmd/brig/commands.glob..func4(0x272df80, 0xc00027a500, 0x1, 0x5, 0x0, 0x0)
	/Users/vdice/go/src/github.com/Azure/brigade/brig/cmd/brig/commands/build_list.go:47 +0x9f
github.com/Azure/brigade/vendor/github.com/spf13/cobra.(*Command).execute(0x272df80, 0xc00027a4b0, 0x5, 0x5, 0x272df80, 0xc00027a4b0)
	/Users/vdice/go/src/github.com/Azure/brigade/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/Azure/brigade/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x272d3a0, 0x272d3a0, 0xc000169f50, 0x0)
	/Users/vdice/go/src/github.com/Azure/brigade/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/Azure/brigade/vendor/github.com/spf13/cobra.(*Command).Execute(0x272d3a0, 0x272d3a0, 0xc000169f88)
	/Users/vdice/go/src/github.com/Azure/brigade/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
	/Users/vdice/go/src/github.com/Azure/brigade/brig/cmd/brig/main.go:11 +0x31
```

Note: not seeing it occur on every project build listing...

Reply:
ah, you're right! fixed, weird that it did not break on my (manual) tests... check it out

Reply:
👍 Looks good now.